### DISCUSSION TITLE
change all day event to specific time

### DISCUSSION THREAD

**Discussion by pupazzooo** [derekantrican/GAS-ICS-Sync#381] [Q&A] [answered]
Hi everyone, I'm trying to change the "all day" events so that we have a specific time (17:00 to 18:00). Is this possible ? I tried to edit icalEvent.startDate in Helpers.gs without result. only errors. can you help me please?

**Comment by jonas0b1011001** [comment]
Give this a shot.
Replace [Helpers.gs L281-289](https://github.com/derekantrican/GAS-ICS-Sync/blob/9990d1b037bbc3e2d72f130f8a815c0f71bd3bcd/Helpers.gs#L281) with this:

```
var tzid = calendarTz;
icalEvent.startDate.isDate = false;
newEvent = {
  start: {
    dateTime : icalEvent.startDate.adjust(0,17,0,0).toString(),
    timeZone : tzid
  },
  end: {
    dateTime : icalEvent.startDate.adjust(0,1,0,0).toString(),
    timeZone : tzid
  }
};
```

**Reply by pupazzooo** [reply]
not work

**Reply by jonas0b1011001** [reply]
You get an error? A bit more detail would be helpful.

You need to sync to a different calendar or remove the all day events from the existing target calendar for the changes to show.

**Reply by pupazzooo** [reply]
in my ics i have only all day event, so the function that the script performs is L255-265. When i insert your code inside the L255-265 i have error. When i insert the line in L281-289 nothing change. I remove the event in my target calendar.

**Reply by pupazzooo** [reply]
plus I have some events that last several days and I still have to start at 5pm and finish at 6pm on the last day (the third)

**Reply by pupazzooo** [reply]
this is the error: 9 nov 2023, 18:24:18	Informazioni	Error, Retrying...GoogleJsonResponseException: API call to calendar.events.insert failed with error: Bad Request

**Reply by jonas0b1011001** [reply, marked as answer]
> in my ics i have only all day event, so the function that the script performs is L255-265.

Seems like you are using an old version, mind updating so we speak of the same version?

> plus I have some events that last several days and I still have to start at 5pm and finish at 6pm on the last day (the third)

```
var tzid = calendarTz;
icalEvent.startDate.isDate = false;
icalEvent.endDate.isDate = false;
newEvent = {
  start: {
    dateTime : icalEvent.startDate.adjust(0,17,0,0).toString(),
    timeZone : tzid
  },
  end: {
    dateTime : icalEvent.endDate.adjust(-1,18,0,0).toString(),
    timeZone : tzid
  }
};
```

**Reply by pupazzooo** [reply]
thank you, all work
